Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -305,14 +319,37 @@ public async Task<string> WalkTree(string treeNodeKey)
                         this.walkTreeCts.Token.ThrowIfCancellationRequested();
                     }
 
-                    await this.Parameters.Callbacks.BeforeVisitNode(
-                        this.Parameters.SessionId,
-                        current,
-                        await this.EvaluateDynamicProperty(this.Schema.Tree[current].Properties, null),
-                        this.Parameters.UserContext,
-                        this.Parameters.TreeName,
-                        this.Parameters.RootSessionId,
-                        this.walkTreeCts.Token).ConfigureAwait(false);
+                    TreeNodeContext treeNodeContext = null;
+                    if (this.Parameters.CallbacksV2 != null)
+                    {
+                        // If applicable, use the new CallbacksV2 with ITreeWalkerCallbacksV2
+                        treeNodeContext = new TreeNodeContext(
+                            this.Parameters.SessionId,
+                            current,
+                            await this.EvaluateDynamicProperty(this.Schema.Tree[current].Properties, null),
+                            this.Parameters.UserContext,
+                            this.walkTreeCts.Token,
+                            this.Parameters.TreeName,
+                            this.Parameters.RootSessionId,
+                            currentNodeSkipActionContext:null
+                        );
+
+                        // Follow the previous Callbacks with ITreeWalkerCallbacks.
```

Comment:
>// Follow the previous Callbacks with ITreeWalkerCallbacks. [](start = 24, length = 59)

Remove this duplicate and misplaced comment.

Reply:
fixed

---
In reply to: [505917480](https://github.com/microsoft/Forge/pull/32#discussion_r505917480) [](ancestors = 505917480)